tests/testthat/test_driver_drake.R

```diff
@@ -0,0 +1,129 @@
+# Test driver_drake
+
+context("driver_drake")
+
+#setup
+devtools::load_all()
+library(drake)
+test_cache <- new_cache()
+xml_dir <- paste0(normalizePath(file.path("../..", XML_DIR), winslash = "/"), "/" )
+
+###########
+ test_that("driver_drake runs with no errors",{
+   expect_error(driver_drake(stop_after = c("module_water_L145.water_demand_municipal"), xmldir = xml_dir, cache = test_cache), NA)
+ })
+
+##########
+ test_that("catches bad input", {
+   expect_error(driver_drake(return_inputs_of = 1))
+   expect_error(driver_drake(return_outputs_of = 1))
+   expect_error(driver_drake(return_data_names = 1))
+   expect_error(driver_drake(return_data_map_only = 1))
+   expect_error(driver_drake(return_plan_only = 1))
+   expect_error(driver_drake(quiet = 1))
+ })
+
+ ###########
+ test_that("plan is a dataframe",{
+   plan <- driver_drake(stop_before = c("module_water_L145.water_demand_municipal"), return_plan_only = TRUE)
+   expect_is(plan, "data.frame")
+  })
+
+
+ ############
+ test_that("load_from_cache works", {
+   data <- load_from_cache(outputs_of("module_water_L145.water_demand_municipal"))
+
+   if(!require("gcamdata.compdata", quietly = TRUE)) {
+     # We couldn't get the "OLD" outputs from the gcamdata.compdata repo
+     # so we will skip this test
+     skip("gcamdata.compdata package not available")
+
+   } else {
+
+     #function to round digits
+     DIGITS <- 3
+     round_df <- function(x, digits = DIGITS) {
+
+       integer_columns <- sapply(x, class) == "integer"
+       x[integer_columns] <- lapply(x[integer_columns], as.numeric)
+
+       numeric_columns <- sapply(x, class) == "numeric"
+       x[numeric_columns] <- round(x[numeric_columns], digits)
+
+       # expect_equivalent no longer accepts rows in different order but we
+       # do want to allow this so we will sort all columns before testing
+       # arrange_columns <- select(x, dplyr::everything())
+       # arrange(x, arrange_columns)
```

Comment:
If not using, delete these comments?

Reply:
Since we want to check equality regardless of order, I changed this test to check if `all.equal()` is true, ignoring row and column order because I was getting this error when using `arrange`  `Error: Argument 1 is of unsupported type data.frame`.  This `round_df()` function is used over in `test_oldnew.R` as well, should we make a similar change there or do you have any other suggestions?

Reply:
Hmm, I don't know. I just meant that it's rarely a good idea to leave commented-out code lying around.